### PR TITLE
WIP: create component-class blueprint

### PR DESCRIPTION
**Pull request by NullVoxPopuli**
Progress on https://github.com/emberjs/ember.js/issues/18275

### CODE REVIEW

**Comment by NullVoxPopuli** [comment]
As is, this could be feature-complete, but I was kinda hoping to be able to move component-class entirely into its own blueprint, so we don't need to keep the two blueprints in sync.

Not sure yet how to do that. 

I see that when you `ember g component`, you get two blueprints:
 - `component`
 - `component-test`

I couldn't figure out how `component-test` was getting thrown in there.

Last todos for this are tests

**Comment by chancancode** [comment]
🙏

File: blueprints/component/index.js [hunk 1]
```diff
@@ -37,6 +37,7 @@ module.exports = {
         { tc: '@ember/component/template-only' },
         { nc: '' },
         { 'no-component-class': '' },
+        { 'with-component-class': '@glimmer/component' },
```
Comment (chancancode) [inline blueprints/component/index.js]: This should be `OCTANE ? '@glimmer/component' : '@ember/component'`

**Comment by rwjblue** [comment]
Closed by https://github.com/emberjs/ember.js/pull/18396

**Pull request closed without merging**